src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/ServiceValueResolver.php

```diff
@@ -41,6 +41,8 @@ public function supports(Request $request, ArgumentMetadata $argument)
             $controller = $controller[0].'::'.$controller[1];
         }
 
+        $controller = ltrim($controller, '\\');
```

Comment:
trim takes a list of characters, thus having two backslashes is redundant. You may want to use `\ltrim` too but not sure about the convention here

Reply:
Hi @Meroje if I don't use the two backslashes, it will consider that I'm escaping the last `'` and never close the string: ```php ltrim($controller, '\')```

Reply:
I'm not sure it's useful for ltrim: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3048

Reply:
It would be great to put the ltrim in a condition.
Basically, I'd suggest doing this:
```php
if (\is_array($controller) && \is_callable($controller, true) && \is_string($controller[0])) {
    $controller = $controller[0].'::'.$controller[1];
} elseif (!\is_string($controller) || '' === $controller) {
    return false;
}
if ('\\' === $controller[0]) {
    $controller = ltrim($controller, '\\');
}

return \$this->container->has($controller) && $this->container->get($controller)->has($argument->getName()); 
```